### PR TITLE
Weightless2

### PR DESCRIPTION
**Pull request by ghost**
[Changelogs]: # Re-adds an event from old hippie where the grav gen temporarily turns off and on. 

:cl:
add: Gravity disruption event.
/:cl:

[why]: # Fun little event pulled from oldschool hippie code

### CODE REVIEW

**Comment by JohnGinnane** [comment]
`did you test this`

**Comment by ghost** [comment]
Yup, it works.

**Review by JohnGinnane — DISMISSED** [review]

**Comment by JohnGinnane** [comment]
Do we need head input for old ports?

File: hippiestation/code/modules/events/weightless.dm [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+/datum/round_event/weightless/setup()
+	startWhen = rand(0,10)
+	endWhen = rand(80,120)
+
+/datum/round_event/weightless/announce()
+	priority_announce("Warning: Failsafes for the station's artificial gravity arrays have been triggered. Please be aware that if this problem recurs it may result in formation of gravitational anomalies. Nanotrasen wishes to remind you that the unauthorised formation of anomalies within Nanotrasen facilities is strictly prohibited by health and safety regulation [rand(99,9999)][pick("a","b","c")]:subclause[rand(1,20)][pick("a","b","c")].")
+
+/datum/round_event/weightless/start()
+	for(var/obj/machinery/gravity_generator/main/station/A)
```
Comment (Carbonhell) [inline hippiestation/code/modules/events/weightless.dm]: Those are in world loops ree

**Review by Carbonhell — DISMISSED** [review]

File: hippiestation/code/modules/events/weightless.dm (hunk 1 above)
Comment (JohnGinnane) [reply hippiestation/code/modules/events/weightless.dm]: ye I think use `in GLOB.machines`

Comment (ghost) [reply hippiestation/code/modules/events/weightless.dm]: Ok will change soon.. the for loop was how the old one was written but that effects the derelict’s grav gen as well. You guys know how to target only the station’s?

Comment (JohnGinnane) [reply hippiestation/code/modules/events/weightless.dm]: @Carbonhell

**Comment by Pyko1** [comment]
+1 needed or not, and no refunds.

**Comment by ghost** [comment]
ok
+1

**Comment by ghost** [comment]
+1

File: hippiestation/code/modules/events/weightless.dm [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+/datum/round_event/weightless/start()
+	for(var/obj/machinery/gravity_generator/main/station/A in GLOB.machines)
+		A.set_state(0)
+	if(control)
+		control.weight *= 2
+
+/datum/round_event/weightless/end()
+	for(var/obj/machinery/gravity_generator/main/station/A in GLOB.machines)
+		A.set_state(1)
```
Comment (JohnGinnane) [inline hippiestation/code/modules/events/weightless.dm]: @Carbonhell @Spankmonkey777 do we need to check `if (A)`? Will this loop iterate through grav gen objects ONLY in the `GLOB.machines` list?

Comment (ghost) [reply hippiestation/code/modules/events/weightless.dm]: Dono, I guess it can't hurt to put a check. I'll do that

**Review by Carbonhell — APPROVED** [review]

**Pull request merged**